src/registrar/signals.py

```diff
@@ -0,0 +1,17 @@
+from django.db.models.signals import post_save
+from django.contrib.auth.models import User
+from django.dispatch import receiver
+
+from .models import UserProfile
+
+
+@receiver(post_save, sender=User)
+def create_profile(sender, instance, created, **kwargs):
+    if created:
+        UserProfile.objects.create(user=instance)
+
+
+@receiver(post_save, sender=User)
+def save_profile(sender, instance, **kwargs):
+    # instance is a User, it has a profile from the one-to-one relation
+    instance.userprofile.save()
```

Comment:
Suggest checking for the existence of `userprofile` and creating it if it does not exist.